src/Microsoft.Graph.Core/Serialization/DerivedTypeConverter.cs

```diff
@@ -241,7 +241,17 @@ public override void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions
                     foreach (var item in additionalData)
                     {
                         writer.WritePropertyName(item.Key);
-                        JsonSerializer.Serialize(writer, item.Value, item.Value.GetType(), options);
+                        // Check if value is null to choose the JsonSerializer.Serialize overload as System.Text.Json no longer supports 
+                        // the type parameter being null. 
+                        // Ref: https://docs.microsoft.com/en-us/dotnet/core/compatibility/serialization/5.0/jsonserializer-serialize-throws-argumentnullexception-for-null-type
+                        if (item.Value == null)
+                        {
+                            JsonSerializer.Serialize(writer, item.Value, options);
+                        }
+                        else
+                        {
+                            JsonSerializer.Serialize(writer, item.Value, item.Value.GetType(), options);
```

Comment:
```suggestion
                            JsonSerializer.Serialize(writer, item.Value, item.Value?.GetType(), options);
```

and remove the if/else

Reply:
Hey @baywet,
The change is also because we can't do that anymore now as now the function now throws an `ArgumentNullException` for that parameter.
See the changelog at this link here. https://docs.microsoft.com/en-us/dotnet/core/compatibility/serialization/5.0/jsonserializer-serialize-throws-argumentnullexception-for-null-type

Reply:
but isn't this passing null?
```CSharp
if (item.Value == null)
{
      JsonSerializer.Serialize(writer, item.Value, options);
}
```

Reply:
Yes. The issue is the third parameter (the Type). So we use an overload that doesn't need the parameter.

Reply:
aaah I see, sorry, morning here, I didn't register the difference in number of parameters somehow.

Reply:
No worries 😄. Thanks for taking a look!